File: ios/Classes/Player/Models/AdvertisementAnalyticsRequest.swift
//
//  AdvertisementAnalyticsRequest.swift
//  udevs_video_player
//
//  Created by Abdurahmon on 01/08/2024.
//

import Foundation

struct AdvertisementAnalyticsRequest{
    let id:String
    let interested:Bool
    let click:Bool
    let viewTime: Int
    
    func toMap()->[String:Any]{
        return [
            "id": id,
            "interested":interested,
            "click":click,
            "view_time":viewTime
        ]
    }
}

File: android/src/main/kotlin/uz/udevs/udevs_video_player/models/advertisement/AdvertisementAnalyticsRequest.kt
package uz.udevs.udevs_video_player.models.advertisement

import com.google.gson.annotations.SerializedName
import java.io.Serializable
data class AdvertisementAnalyticsRequest(
    val id: String,
    val click: Boolean,
    val interested: Boolean,
    @SerializedName("view_time")
    val viewTime: Int,
): Serializable